### PR TITLE
vertical padding is borked on new attachment UI after going into file selector and out again

### PR DESCRIPTION
**Pull request by gileluard**
fix https://github.com/vector-im/element-ios/issues/4156

### CODE REVIEW

**Comment by gileluard** [comment]
As an FYI, the only issue here is that the detection of the external keyboard doesn't work anymore. This happens typically in a view controller after having presented another view controller.

There are some discussions [here](https://stackoverflow.com/questions/31991873/how-to-reliably-detect-if-an-external-keyboard-is-connected-on-ios-9). I don't like this fix but it's the only reliable one I found.

**Review by manuroe — APPROVED** [review]
I created a dedicated issue for external keyboard: https://github.com/vector-im/element-ios/issues/4180

File: CHANGES.rst [hunk 1]
```diff
@@ -8,7 +8,8 @@ Changes to be released in next version
  * 
 
 🐛 Bugfix
- * 
+ * vertical padding is borked on new attachment UI after going into file selector and out again (vector-im/element-ios/issues/4156).
+ * vertical layout of typing notifs can go wonky (vector-im/element-ios/issues/4159).
```
Comment (ismailgulek) [inline CHANGES.rst:12]:
```suggestion
 * Vertical padding is borked on new attachment UI after going into file selector and out again (vector-im/element-ios/issues/4156).
 * Vertical layout of typing notifs can go wonky (vector-im/element-ios/issues/4159).
```

File: MatrixKit/Controllers/MXKViewController.m [hunk 2]
```diff
@@ -638,13 +638,9 @@ - (void)onKeyboardWillShow:(NSNotification *)notif
 
     // Detect if an external keyboard is used by checking that
     // the bottom of the provided keyboard frame is outside of the screen
```
Comment (ismailgulek) [inline MatrixKit/Controllers/MXKViewController.m:640]: These comments became obsolete. Can you remove them?

File: MatrixKit/Controllers/MXKViewController.m [hunk 3]
```diff
@@ -638,13 +638,9 @@ - (void)onKeyboardWillShow:(NSNotification *)notif
...
-    BOOL hasExternalKeyboard = NO;
     CGRect keyboard = [self.view convertRect:endRect fromView:self.view.window];
     CGFloat height = self.view.frame.size.height;
-    if ((keyboard.origin.y + keyboard.size.height) > height)
-    {
-        hasExternalKeyboard = YES;
-    }
+    BOOL hasExternalKeyboard = keyboard.size.height <= 80;
```
Comment (ismailgulek) [inline MatrixKit/Controllers/MXKViewController.m:643]: It's a shame we need to check external keyboard like this, thanks Apple for that. Can you just move this value `80` to a constant please?

**Review by ismailgulek — APPROVED** [review]

**Pull request merged**